synapse/lib/stormtypes.py

```diff
@@ -5293,6 +5306,10 @@ async def _methUserGrant(self, iden):
         self.runt.confirm(('auth', 'user', 'grant'))
         await self.runt.snap.core.addUserRole(self.valu, iden)
 
+    async def _methUserGrants(self, idens):
+        self.runt.confirm(('auth', 'user', 'grant'))
```

Comment:
Should we also check whether they have `auth.user.revoke` permission since we are replacing the old roles?

Reply:
good find

Reply:
Or... even better, we could intersect the old roles and the new ones into adds/dels and decide which perms you need?

Reply:
doing the intersection requires a check and set which could change. testing both grant and revoke permissions would capture all cases?